prov/acmp/src/acmp.c

```diff
@@ -1774,6 +1845,7 @@ acmp_resolve_dest(struct acmp_ep *ep, struct acm_msg *msg, uint64_t id)
 		acm_log(2, "have address, resolving route\n");
 		acm_increment_counter(ACM_CNTR_ADDR_CACHE);
 		atomic_inc(&ep->counters[ACM_CNTR_ADDR_CACHE]);
+		acmp_acquire_sgid(saddr, dest);
```

Comment:
Please also look at the other places where acmp_resolve_path_sa() is invoked and determine how the SGID is set in those locations.